### PR TITLE
Solo study correction

### PR DESCRIPTION
**Pull request by Ovoland**
I corrected the Buttons of the solo study screen so that all different icons and texts are correctly displayed. Then I added the "route" to the timer when clicking on the icon. As the timer had issues, I did small corrections so that the basic functionality can be correctly used. However, further and deeper correction are needed as well as tests to be sure that the errors won't come again. Finally I generalized the UI code structure. Now the screen are either implemented using DrawerMenu or a Scaffold.

### CODE REVIEW

**Comment by RaulinN** [comment]
Make sure you guys review #71 asap if you think the PR is ready. There's a huge number of files changed. The more you wait to merge this PR, the more painful the merge conflicts will be. For instance, this PR is now 10x as big because there's the work of #78 and #71 which also makes it very complicated to review

**Comment by hanelmouta** [comment]
everything looks good for me , checked the route to the timer and the changes to UI

**Review by hanelmouta — APPROVED** [review]
I approve of the changes made

**Review by TdVBd — APPROVED** [review]
Great job overall, the SoloStudy buttons do not seem to be working, altough it might be because the viewmodel has not been implemented yet

**Comment by Ovoland** [comment]
I have fixed the different conflict with the main branch, I changed the name of the DrawerMenu to MainScreenScaffold to make it more understandable and I resize the icons and text of the timer screen

File: app/src/main/java/com/github/se/studybuddies/viewModels/TimerViewModel.kt [hunk 1]
```diff
@@ -33,7 +42,9 @@ class TimerViewModel : ViewModel() {
           viewModelScope1.launch {
             while (isActive) {
               delay(1000)
-              _timer.value--
+              if (_timer.value > 0) _timer.value-- else _timerEnd.value = true
+              delay(1000)
+              _timerEnd.value = false
             }
           }
```
Comment (RaulinN) [inline app/src/main/java/com/github/se/studybuddies/viewModels/TimerViewModel.kt:49]: Océane: M2 comment

**Pull request merged**